mobile/src/main/java/org/openhab/habdroid/ui/ConnectionWebViewClient.kt

```diff
@@ -48,6 +51,18 @@ open class ConnectionWebViewClient(
         handler.proceed(connection.username, connection.password)
     }
 
+    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
+        if (request.url.toString().toHttpUrlOrNull()?.resolve("/") == view.url?.toHttpUrlOrNull()?.resolve("/")) {
```

Comment:
And maybe also check whether both `isForMainFrame()` and `hasGesture()` returns true before redirecting to external browser?

Reply:
> It looks like you want to compare whether the host is the same only, is this correct? 

Not exactly. I want to check if scheme, host and port are equal. There could be another application running on the same host, but a different port.

Reply:
> And maybe also check whether both isForMainFrame() and hasGesture() returns true before redirecting to external browser?

With the cell shown above both functions return true. IMO external pages should be opened in a browser to avoid sending credentials.

Reply:
> With the cell shown above both functions return true

AFAICT both things (URL and load reason) are orthogonal. I just want to avoid any potential internal loads suddenly opening the browser.

Reply:
I checked https://demo.openhab.org/#!/page/positions in my desktop browser: The tiles are downloaded whenever the position change, but in the app this doesn't even trigger `shouldOverrideUrlLoading()`.
IMO we will see in the beta if there any new issues with this change.